From 34aba4f12d47866c7edb283d3b57d26cbe133668 Mon Sep 17 00:00:00 2001
From: Numan Siddique <nusiddiq@redhat.com>
Date: Thu, 8 Dec 2016 21:03:09 +0530
Subject: [PATCH] Setting inactivity probes for OVN NB and DB servers

This patch adds additional configuration setting for to set
inactivtiy probe intervals for OVN NB and DB servers

Change-Id: I01fa4fff4b06417247c968eeb34148ecc92bc6bb
---
 environments/neutron-ml2-ovn.yaml |  2 ++
 puppet/services/ovn-dbs.yaml      | 16 ++++++++++++++++
 2 files changed, 18 insertions(+)

diff --git a/environments/neutron-ml2-ovn.yaml b/environments/neutron-ml2-ovn.yaml
index ec51239..fde4590 100644
--- a/environments/neutron-ml2-ovn.yaml
+++ b/environments/neutron-ml2-ovn.yaml
@@ -26,3 +26,5 @@ parameter_defaults:
   NeutronServicePlugins: 'qos,networking_ovn.l3.l3_ovn.OVNL3RouterPlugin'
   NeutronVniRanges: ['1:65536', ]
   OVNBridgeMappings: 'datacentre:br-ex'
+  OVNNorthboundInactivityProbe: 180000
+  OVNSouthboundInactivityProbe: 180000
diff --git a/puppet/services/ovn-dbs.yaml b/puppet/services/ovn-dbs.yaml
index 302628d..5e8ed6f 100644
--- a/puppet/services/ovn-dbs.yaml
+++ b/puppet/services/ovn-dbs.yaml
@@ -26,6 +26,20 @@ parameters:
     description: Port of the OVN Southbound DB server
     type: number
     default: 6642
+  OVNNorthboundInactivityProbe:
+    description: Maximum number of milliseconds of idle time on connection to
+                 the neutron ML2 OVN driver/ovn-northd before OVN Northbound
+                 DB server sends an inactivity probe message.
+    type: number
+    default: 180000
+  OVNSouthboundInactivityProbe:
+    description: Maximum number of milliseconds of idle time on connection to
+                 the neutron ML2 OVN driver/ovn-northd/ovn-controller before
+                 OVN Southbound DB server sends an inactivity probe
+                 message.
+    type: number
+    default: 180000
+
 
 outputs:
   role_data:
@@ -36,5 +50,7 @@ outputs:
           ovn::northbound::port: {get_param: OVNNorthboundServerPort}
           ovn::southbound::port: {get_param: OVNSouthboundServerPort}
           ovn::northd::dbs_listen_ip: {get_param: [ServiceNetMap, OvnDbsNetwork]}
+          ovn::northd::ovn_northbound_inactivity_probe: {get_param: OVNNorthboundInactivityProbe}
+          ovn::northd::ovn_southbound_inactivity_probe: {get_param: OVNSouthboundInactivityProbe}
       step_config: |
         include ::tripleo::profile::base::neutron::ovn_northd
-- 
2.9.3

